sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -579,6 +579,22 @@ class CodegenContext {
     s"${fullName}_$id"
   }
 
+  /**
+   * Creates an `ExprValue` representing a local java variable of required data type.
+   */
+  def freshName(name: String, dt: DataType): VariableValue = JavaCode.variable(freshName(name), dt)
```

Comment:
What about some better name like `getLocalVariable` or something similar? this is not just a fresh name...

Reply:
Sounds good. Not sure if @cloud-fan wants me to change it in this PR or other https://github.com/apache/spark/pull/21537#discussion_r194841055.

Reply:
I think either we decide to postpone the introduction of this API to a followup PR or we can change its name now that we are introducing it...

Reply:
how about we move it to `JavaCode` and follow the name convention there and call it `freshVariable`?

Reply:
Putting it in `CodeGenerator` is just for convenience that we can save a `ctx` parameter. I'm fine to let it in `JavaCode` too if you don't think it's verbose.

Reply:
oh I missed the ctx parameter thing, let's leave it.